From 32ec754dba233caefc81431be6c43a2059e44cbf Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 19 Oct 2025 23:35:15 +0300
Subject: [PATCH 1/1] amd64 config STRICT_DEVMEM=n to enable mmap on trixie

Marvell and many other platforms are using mmap() sys-call.
Kernel-6.12 disables/prohibits it by dfault, mmap() just retuns NULL.
Re-enable it for proper work by the CONFIG_STRICT_DEVMEM=n.

Done for amd64 only since arm64 also having default
 CONFIG_STRICT_DEVMEM=y does not prohibit the mmap() access.

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 config.local/amd64/config.sonic | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/config.local/amd64/config.sonic b/config.local/amd64/config.sonic
index 1fc546a..ae8dabb 100644
--- a/config.local/amd64/config.sonic
+++ b/config.local/amd64/config.sonic
@@ -158,3 +158,7 @@ CONFIG_AMD_MEM_ENCRYPT=n
 # Unset X86_PAT according to Broadcom's requirement
 CONFIG_X86_PAT=n
 CONFIG_MLXSW_PCI=n
+
+# For marvell-prestera
+CONFIG_STRICT_DEVMEM=n
+
-- 
2.34.1

